src/main/java/ca/on/oicr/gsi/dimsum/data/ProjectSummary.java

```diff
@@ -0,0 +1,53 @@
+package ca.on.oicr.gsi.dimsum.data;
+
+import static java.util.Objects.requireNonNull;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class ProjectSummary {
+  private final String name;
+  private final Map<String, Integer> counts;
```

Comment:
We'll want a stricter data model to enforce a specific set of keys. This also will not convert to a nice JSON representation. Ideal implementation will use specific `int` fields such as `receiptPendingQcCount`, `receiptCompletedCount`, `extractionPendingCount`, etc.

You could add a `addCounts(ProjectSummary.Builder)` method to `ProjectSummary.Builder` to make it easy to combine counts. Then in `CaseLoader`, you'd use a `ProjectSummary.Builder caseSummary` to add up the case counts, and `projectSummary.addCounts(caseSummary)` to push those counts into the project

Reply:
Would adding the counts within the builder be feasible? From my understanding of your suggestion I was thinking of something like `this.totalTestCount` += `caseSummary.totalTestCount`, but this would modify the field of existing `ProjectSummary` object.

Reply:
Yes, counts would all have to be added within the builder. Sorry I wasn't clear, but in my example of `projectSummary.addCounts(caseSummary)`, `projectSummary` and `caseSummary` were both meant to be builders

Reply:
Ah I think I see what you mean. So just to clarify, we would keep track of the builder objects (add the counts if necessary) and push those to the `ProjectSummariesbyName` when we iterate through all the cases?

Reply:
`CaseData.projectSummariesByName` would still be a `Map<String, ProjectSummary>`.

In `CaseLoader#calculateProjectSummaries`, you would have a `Map<String, ProjectSummary.Builder>` to track the project summaries temporarily. Each case would be converted to a `ProjectSummary.Builder` which I'm called `caseSummary`, then for each project in the case, you'd get the project summary from the map (or add new if necessary), and add the caseCounts to it. Once all cases are handled, you'd build all of the `ProjectSummary`s

Reply:
Thanks for clarification